src/fontra/views/editor/panel-selection-transformation.js

```diff
@@ -0,0 +1,534 @@
+    if (glyphController) {
+      await this._setupSelectionInfoHandlers(glyphName);
+    }
+  }
+
+  _getSelectedBounds(layerGlyphController, layerGlyph, pointIndices, componentIndices) {
+    const selectionRects = [];
+    if (pointIndices.length) {
+      const selRect = rectFromPoints(
+        pointIndices.map((i) => layerGlyph.path.getPoint(i)).filter((point) => !!point)
+      );
+      if (selRect) {
+        selectionRects.push(selRect);
+      }
+    }
+
+    for (const componentIndex of componentIndices) {
+      const component = layerGlyphController.components[componentIndex];
+      if (!component || !component.controlBounds) {
+        continue;
+      }
+      selectionRects.push(component.controlBounds);
+    }
+
+    if (selectionRects.length) {
+      const selectionBounds = unionRect(...selectionRects);
+      return selectionBounds;
+    }
+  }
+
+  _getPinPoint(
+    layerGlyphController,
+    layerGlyph,
+    pointIndices,
+    componentIndices,
+    originX,
+    originY
+  ) {
+    let bounds = this._getSelectedBounds(
+      layerGlyphController,
+      layerGlyph,
+      pointIndices,
+      componentIndices
+    );
+    if (!bounds) {
+      bounds = {
+        xMin: 0,
+        xMax: layerGlyph.xAdvance,
+        yMin: 0,
+        yMax: layerGlyph.yAdvance ? layerGlyph.yAdvance : 0,
+      };
+    }
+    const width = bounds.xMax - bounds.xMin;
+    const height = bounds.yMax - bounds.yMin;
+
+    // default from center
+    let pinPointX = bounds.xMin + width / 2;
+    let pinPointY = bounds.yMin + height / 2;
+
+    if (typeof originX === "number") {
+      pinPointX = originX;
+    } else if (originX === "left") {
+      pinPointX = bounds.xMin;
+    } else if (originX === "right") {
+      pinPointX = bounds.xMax;
+    }
+
+    if (typeof originY === "number") {
+      pinPointY = originY;
+    } else if (originY === "top") {
+      pinPointY = bounds.yMax;
+    } else if (originY === "bottom") {
+      pinPointY = bounds.yMin;
+    }
+
+    return { x: pinPointX, y: pinPointY };
+  }
+  _getPointIndicesInclOffCurves(layerGlyph, pointIndices) {
+    if (!pointIndices || pointIndices.length < 1) {
+      return [];
+    }
+    let newPointIndices = new Set();
+    const behaviorFactory = new EditBehaviorFactory(
+      layerGlyph,
+      this.sceneController.selection,
+      this.sceneController.experimentalFeatures.scalingEditBehavior
+    );
+
+    const editBehavior = behaviorFactory.getBehavior("default");
+    const contours = behaviorFactory.contours;
+    for (const [i, arrayPointsIndices] of enumerate(
+      editBehavior.participatingPointIndices
+    )) {
+      if (!arrayPointsIndices) {
+        continue;
+      }
+      arrayPointsIndices.forEach((item) =>
+        newPointIndices.add(item + contours[i].startIndex)
+      );
+    }
+    return Array.from(newPointIndices).sort((a, b) => a - b);
+  }
+
+  async _transformLayerGlyph({
+    originX = this.originX,
```

Comment:
It seems you never set this value from the caller.

Also, I suggest to pass a Transform object instead of all separate parameters. To set the pinpoint for a transform, you can do something like this:
```
    t = Transform().translate(pinX, pinY)
    t = t.transform(theOtherTransform)
    t = t.translate(-pinX, -pinY)
```